### PR TITLE
[Play] - Phase Results

### PR DESCRIPTION
**Pull request by drewjhart**
This PR adds the Phase Results screen for Phase 1 and Phase 2. Attention was put into ensuring that mock data has properly handled in both the detemination of answer types (to assign styling for correct, selected, player correct states) and in the use of helper functions from the `@networking` layer to calculate the score.

**Change Description:**
Changes are broken down into two main areas: 

1. PhaseResults Addition
- This is the bulk of the update and provides the page through `PhaseResult.tsx` as well as child components of `ResultsCard.tsx` and `ResultSelector.tsx`

2. File structure
- I've adjusted some of the way we are organizing our `styledcomponents` under `lib` this has created some changes in files that aren't related to `PhaseResults` but that following the new file paths. Apologies for the clutter in this PR.

**Build:**
I have built this to [play.rightoneducation.com](play.rightoneducation.com) for testing purposes.

**Screenshots:**
image
image

**Figma Reference:**
image
[Figma Link](https://www.figma.com/file/PadzU8GqyhFLCjIt7rQkpE/RightOn-Apps?node-id=13554-169664&t=r3kmWYhOW4ZI5OpS-4)

**Chromatic Build:**
Updated storybook is [here](https://64220218d457f84ba4546eb1-twsusdrfmj.chromatic.com/)

Additions are:
- Pages: PhaseResults
- Molecules: ResultSelector

### CODE REVIEW

File: play/src/pages/PhaseResults.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  ) => {
+    const newScore = ModelHelper.calculateBasicModeScoreForQuestion(
+      session,
+      question,
+      team
+    );
+    return newScore;
+  };
+
+  return (
```
Comment (maniramezan) [inline play/src/pages/PhaseResults.tsx:113]: Great job on breaking it into smaller components, this looks super clean and easy to read

File: play/src/containers/GameSessionContainer.tsx [hunk 2]
```diff
@@ -3,24 +3,60 @@ import {
...
+          teamId="2d609343-de50-4830-b65e-71eb72bb9bef"
+        />
+      );
+    default:
+      return (
+        <GameInProgress
+          {...gameSession}
+          teamAvatar={teamAvatar}
+          teamId="2d609343-de50-4830-b65e-71eb72bb9bef"
+        />
```
Comment (maniramezan) [inline play/src/containers/GameSessionContainer.tsx:59]: Not necessary for this PR, but we'd probably wanna check and validate the varaiebls are not `nil` and are available here for each view, like `gameSession` or `teamAvatar` and etc.

**Review by maniramezan — APPROVED** [review]

File: play/src/components/ResultSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+          <ResultSelectorCorrect>{resultContents}</ResultSelectorCorrect>
+        </Box>
+      );
+    case AnswerState.PLAYER_CORRECT:
+      return (
+        <Box>
+          <Box sx={{ position: 'relative', height: 0 }}>
+            <CorrectStarsStyled
+              src={CorrectStars}
+              alt=""
```
[the remark this answers is not in the document]
Comment (drewjhart) [reply play/src/components/ResultSelector.tsx:131]: nice, fixed.

File: play/src/lib/PlayModels.tsx [hunk 4]
```diff
@@ -12,9 +12,11 @@ import Monster4 from '../img/Monster4.svg';
...
 
 export enum AnswerState {
-  DEFAULT = 'DEFAULT',
-  CORRECT = 'CORRECT',
-  SELECTED = 'SELECTED',
+  DEFAULT = 'DEFAULT', // any answer not correct or selected by player
+  CORRECT = 'CORRECT', // correct answer per gameSession object
+  PLAYER_CORRECT = 'PLAYER_CORRECT', // player has chosen correct answer
+  SELECTED = 'SELECTED', // answer player has selected
+  PREVIOUS = 'PREVIOUS', // answer player has selected in previous phase
```
[the remark this answers is not in the document]
Comment (drewjhart) [reply play/src/lib/PlayModels.tsx:25]: great comment. I've updated all enums in PlayModels and already see the benefit from a dev perspective

File: play/src/pages/PhaseResults.tsx (hunk 1 above)
Comment (drewjhart) [reply play/src/pages/PhaseResults.tsx:113]: nice!

File: play/src/containers/GameSessionContainer.tsx (hunk 2 above)
Comment (drewjhart) [reply play/src/containers/GameSessionContainer.tsx]: yup, great idea. I'll add that to the subscription/routing pr. Thanks again for the reviews!

**Pull request merged**